packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -1523,4 +1367,213 @@ extension FlutterVmService on vm_service.VmService {
     );
     await onRunnable;
   }
+
+  Future<Map<String, dynamic>> flutterDebugDumpApp({
```

Comment:
I avoided extensive refactoring here and mostly copied as is, with the addition of the `isolateId` field